### PR TITLE
DSS-PEPPER-787 add upload file support to the onc history tab

### PR DESCRIPTION
**Pull request by GiCharkviani**
[PEPPER-787](https://broadworkbench.atlassian.net/jira/software/projects/pepper/boards/164?assignee=621677042d057500723f90e4&selectedIssue=PEPPER-787)

1. Added onc history upload page
2. increased allowed characters from 200 to 1000 in the "Location of PX" input field

Screenshot 2023-06-16 at 18 09 31

[PEPPER-787]: https://broadworkbench.atlassian.net/browse/PEPPER-787?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ddp-workspace/projects/ddp-dsm-ui/src/app/oncHistoryUpload/oncHistoryUpload.component.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            console.log(error);
+            this.uploadStatus = UploadStatus.FAIL;
+            this.errorMessage = error.error;
+          }
+        },
+      });
+  }
+
+  fileSelected(file: File): void {
+    if (file.size > 31457280) {
```
Comment (ssettipalli) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/oncHistoryUpload/oncHistoryUpload.component.ts:55]: wonder if size can be configurable but ok for now.

**Review by ssettipalli — APPROVED** [review]
Looks ok to me from my limited angular skill.

File: ddp-workspace/projects/ddp-dsm-ui/src/app/oncHistoryUpload/oncHistoryUpload.component.ts (hunk 1 above)
Comment (amimahd) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/oncHistoryUpload/oncHistoryUpload.component.ts:55]: ,hi  tAnks

File: ddp-workspace/projects/ddp-dsm-ui/src/app/services/session.service.ts [hunk 2]
```diff
@@ -35,6 +36,10 @@ export class SessionService {
     return this.isLoggedIn;
   }
 
+  public get selectedRealm(): string {
```
Comment (otchet-broad) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/services/session.service.ts:39]: Thank you

**Review by otchet-broad — APPROVED** [review]
LGTM

File: ddp-workspace/projects/ddp-dsm-ui/src/app/oncHistoryUpload/oncHistoryUpload.component.ts (hunk 1 above)
Comment (amimahd) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/oncHistoryUpload/oncHistoryUpload.component.ts:55]: rmed somewhere
      this.uploadStatus = UploadStatus.FAIL;
      this.errorMessage = 'File size should be less than 30 MB';
    } else {
      this.uploadStatus = UploadStatus.UPLOADING;
      this.uploadFile(file);
    }
  }

}

// Upload status enum
enum UploadStatus {
  UPLOADING,
  SUCCESS,
  FAIL,
}
=== stderr ===

com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 3 column 2 path $
	at com.google.gson.JsonParser.parse(JsonParser.java:8)
	at com.google.gson.JsonParser.parse(JsonParser.java:1)
	at xf.a(CompilerHelper.java:15)
	at xs$b$a$a.doInBackground(ExecuteDialog.java:7)
	at android.os.AsyncTask$3.call(AsyncTask.java:378)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:919)
Caused by: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 3 column 2 path $
	at com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:3)
	at com.google.gson.stream.JsonReader.checkLenient(JsonReader.java:2)
	at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:18)
	at com.google.gson.stream.JsonReader.peek(JsonReader.java:2)
	at com.google.gson.JsonParser.parse(JsonParser.java:4)
	... 6 more

**Pull request merged**